metadata_extract/text.py

```diff
@@ -17,29 +18,67 @@ def append_to_context(self, extra_context: str) -> None:
         self.context = (self.context or '') + extra_context
 
 
-files = InitFiles()
-LABELS = files.get_labels()
-STOPWORDS = files.get_stopwords()
+def __labels() -> dict[str, str]:
+    return ResourceLoader.get_labels()
+
 
 ISXN_PATTERN = {
     'ISSN': re.compile(r"\D(\d{4}[–-][\dX]{4})\D"),
     'ISBN': re.compile(r"\D([\d–-]{13,17})\D")
 }
-REPORT_PATTERN = re.compile(fr'^(\w+)\W({LABELS["report"]})\W', re.IGNORECASE)
-TYPE_PATTERN_1 = re.compile(fr'\b({LABELS["reportType"]})\b', re.IGNORECASE)
-TYPE_PATTERN_2 = re.compile(r'\bNOU\b')
-DOC_TYPE_MAPPING = files.get_doc_type_mapping()
-PUBLISHER_LABEL = re.compile(fr'({LABELS["publisher"]}):?', re.IGNORECASE)
-NO_LETTERS_PATTERN = re.compile(r'^[\W\d]+$')
-AUTHOR_LABEL = re.compile(fr'({LABELS["author"]}):?', re.IGNORECASE)
-NAME_PATTERN = regex.compile(r"\b[^\P{Lu}][^\P{Ll}]*[-|‐]?[^\P{Lu}]?[^\P{Ll}’]*\.?" +
-                             r"(?: [^\P{Lu}][^\P{Ll}’]*[-|‐]?[^\P{Lu}]?[^\P{Ll}]*\.?)+\b(?! *\()")
-PARENTHESIS_PATTERN = re.compile(r"\(.*?\)")
-DOUBLE_CAPITAL_LETTER_PATTERN = re.compile(r"\b[A-Z]{2,}\b")
-BINDING_WORD_PATTERN = re.compile(fr'\b(?:{LABELS["bindingWords"]})\b|&|,')
-SPECIAL_CHAR_AND_BINDING_PATTERN = re.compile(fr'[;:,.]|({LABELS["bindingWords"]})\b|&+')
-NON_ALPHANUMERIC_PATTERN = re.compile(r"\W+")
-PHOTOGRAPH_LABEL = re.compile(fr'\b({LABELS["photo"]})\b', re.IGNORECASE)
+
+
+def report_pattern() -> Pattern[str]:
+    return re.compile(fr'^(\w+)\W({__labels()["report"]})\W', re.IGNORECASE)
```

Comment:
The advantage of using `re.compile` is that the resulting object is more efficient for text matching. But we lose this advantage if we rebuild the regex object at every call of `report_pattern`... One solution is:

```suggestion
REPORT_PATTERN = None

def report_pattern() -> Pattern[str]:
    global REPORT_PATTERN
    if not REPORT_PATTERN:
        REPORT_PATTERN = re.compile(fr'^(\w+)\W({__labels()["report"]})\W', re.IGNORECASE)
    return REPORT_PATTERN
```

The use of `global` objects is maybe not the most elegant, but it's one way to avoid defining a `Text` class.

Reply:
One way to avoid many `global` is to just have one global dict:

```suggestion
PATTERNS = {}

def report_pattern() -> Pattern[str]:
    if 'REPORT' not in PATTERNS:
        PATTERNS['REPORT'] = re.compile(fr'^(\w+)\W({__labels()["report"]})\W', re.IGNORECASE)
    return PATTERNS['REPORT']
```